src/main/java/com/vinylteam/vinyl/web/templater/PageGenerator.java

```diff
@@ -63,28 +73,44 @@ private Context getContext(List<Vinyl> list, Map<String, String> attributes) {
             context.setVariable("message", message);
         }
 
-        context.setVariable("vinylList", list);
-        if (!list.isEmpty()) {
-            context.setVariable("firstVinyl", list.get(0));
+        //for catalog page
+        context.setVariable("vinylList", vinylList);
+
+        //for search & vinyl pages
+        if (!vinylList.isEmpty()) {
+            context.setVariable("firstVinyl", vinylList.get(0));
         }
-        if (list.size() > 1) {
-            if (list.size() >= 5) {
+
+        //for search page
+        if (vinylList.size() > 1) {
+            if (vinylList.size() >= 5) {
                 for (int i = 1; i < 5; i++) {
-                    firstVinylRow.add(list.get(i));
+                    firstVinylRow.add(vinylList.get(i));
                 }
             } else {
-                for (int i = 1; i < list.size(); i++) {
-                    firstVinylRow.add(list.get(i));
+                for (int i = 1; i < vinylList.size(); i++) {
+                    firstVinylRow.add(vinylList.get(i));
                 }
             }
             context.setVariable("firstVinylRow", firstVinylRow);
         }
-        if (list.size() > 5) {
-            for (int i = 5; i < list.size(); i++) {
-                otherVinylRow.add(list.get(i));
+        if (vinylList.size() > 5) {
+            for (int i = 5; i < vinylList.size(); i++) {
+                otherVinylRow.add(vinylList.get(i));
             }
             context.setVariable("otherVinylRow", otherVinylRow);
         }
+
+        //for vinyl page
```

Comment:
For comments in code please use:     /**      */
Check examples it the existing libraries.